File: GalleryApp/src/main/java/ViewModel/Commands/GalleryCommand.java
package ViewModel.Commands;

public class AddArtistCommand implements ICommands {
    private final Runnable action;

    public AddArtistCommand(Runnable action) {
        this.action = action;
    }

    @Override
    public void execute() {
        if (action != null) {
            action.run();
        }
    }
}

File: AssignmentMVVM/ViewModel/Commands/GalleryCommand.cs
using System.Windows.Input;

namespace AssignmentMVVM.ViewModel.Commands
{
    public class GalleryCommand : ICommand
    {
        private readonly Action action;

        public event EventHandler? CanExecuteChanged;

        public GalleryCommand(Action action)
        {
            this.action = action;
        }

        public bool CanExecute(object parameter) => true;

        public void Execute(object parameter)
        {
            action();
        }
    }
}